packages/react-tinacms-editor/src/tinacms-inline/inline-wysiwyg.tsx

```diff
@@ -42,14 +42,20 @@ export function InlineWysiwyg({
     if (!passedInImageProps) return
     return {
       async upload(files: File[]) {
-        const allMedia = await cms.media.store.persist(
-          files.map(file => ({
-            directory: passedInImageProps?.directory || '',
-            file,
-          }))
-        )
+        const filesToUpload = files.map(file => ({
+          directory: passedInImageProps?.directory || '',
+          file,
+        }))
+
+        const allMedia = await cms.media.store.persist(filesToUpload)
 
-        return allMedia.map(media => passedInImageProps.parse(media.filename))
+        return allMedia.map(media => {
+          if (passedInImageProps.parse) {
+            return passedInImageProps.parse(media.filename)
+          } else {
+            return media.filename
```

Comment:
We are essentially creating a default parse function here. What if we prepended the directory here by default?